### PR TITLE
doc: adding reviewer guidelines

### PR DESCRIPTION
**Pull request by asettle**
Signed-off-by: Alexandra Settle <asettle@suse.com>

### CODE REVIEW

**Review by jhesketh — DISMISSED** [review]
Great start, thanks :-).

I think we should separate out `reviewing pull requests` and `merging pull requests`.

`reviewing pull requests` should cover what to check for (it's probably self explanatory so can likely be short).

`merging pull requests` should cover what is required before a PR is merged, but also when to merge a PR.
For example: All CI has run and is passing, at least 1 positive review/approval, open for feedback for at least 24 hours, no outstanding requests for changes, if still open after $days okay to self-merge? etc.
This should also clarify the needs-rebase case hopefully.

(just my 2cents anyway)

File: CONTRIBUTING.md
Comment (ml8mr) [inline CONTRIBUTING.md:37]: IMO, the primary function of reviews is to ensure quality. However, we also want PRs to be reviewed quickly, to avoid the blocking problem.

Comment (jecluis) [reply CONTRIBUTING.md:37]: Agree with @ml8mr

File: CONTRIBUTING.md
Comment (jecluis) [inline CONTRIBUTING.md:49]: I'd argue that even draft PRs should be reviewed if a review is requested.

IMO, if a PR is opened is because the author wants/needs feedback, even if it's a draft. Same goes for WIP.

**Review by jecluis — COMMENTED** [review]
I don't agree with not reviewing PRs based on draft or WIP. A PR with a request for review should always be reviewed. The whole point of drafts or WIP (IMO) is to make it clear the work is not final, but that should not discourage (heck, it should encourage!) reviews.

File: CONTRIBUTING.md
Comment (l-mb) [reply CONTRIBUTING.md:49]: The keyword is "require"; draft PRs don't need one (but can request feedback). But then, draft/WIP won't get merged, and will require a mandatory review prior to merge, hence this may be redundant. Maybe rephrase paragraph to:

```suggestion
If your pull request is not yet in the state where it can be meaningfully reviewed and merged, please leave it in draft status or tag with the `WIP` label. You can of course still ask for feedback by manually assigning a reviewer or pinging them directly at this stage.
```

**Review by l-mb — DISMISSED** [review]
LGTM, possibly with the one paragraph altered as discussed (but not a blocking requirement from my side).

**Review by l-mb — APPROVED** [review]
LGTM.

**Review by jecluis — APPROVED** [review]

**Review by s0nea — DISMISSED** [review]

**Review by ml8mr — APPROVED** [review]

File: CONTRIBUTING.md
Comment (jhesketh) [inline CONTRIBUTING.md:73]: Can you please clarify if this means we should not merge a PR unless we've discussed it with the author. And otherwise it is up to the author to merge their own pull request after approvals?

**Review by l-mb — APPROVED** [review]
Let's get this merged.

**Review by s0nea — APPROVED** [review]

**Pull request merged**
